pkg/controllers/configobservation/console/observe_consoleurl.go

```diff
@@ -20,8 +23,23 @@ func ObserveConsoleURL(genericlisters configobserver.Listers, recorder events.Re
 	listers := genericlisters.(configobservation.Listers)
 	errs := []error{}
 
+	clusterVersionConfig, err := listers.ClusterVersionLister.Get("cluster")
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	isConsoleCapabilityEnabled := false
+	for _, capability := range clusterVersionConfig.Status.Capabilities.EnabledCapabilities {
+		if capability == configv1.ClusterVersionCapabilityConsole {
+			isConsoleCapabilityEnabled = true
+		}
+	}
+
 	consoleConfig, err := listers.ConsoleLister.Get("cluster")
 	if err != nil {
+		if errors.IsNotFound(err) && isConsoleCapabilityEnabled {
```

Comment:
I think you want `!isConsoleCapabilityEnabled` instead of `isConsoleCapabilityEnabled` here, for "we can't find the `cluster` Console, but that's ok, because this is a console-less cluster".

I'm a bit unsure about `errors.IsNotFound(err)`.  Does that match the errors we get when requesting a resource whose backing CRD is missing?

Reply:
Thats a fair point @wking ... Not entirely sure what should be the other then the `IsNotFound` error